src/app/services/connectors/rest-api/rest-api-connector.service.ts

```diff
@@ -1115,6 +1115,33 @@ export class RestApiConnectorService extends ApiConnector {
     };
   }
 
+  /**
+   * Factory to create a STIX object validator (POST) function
+   * @template T the type to validate
+   * @param {AttackType} attackType the type to validate
+   * @returns validator (POST) function
+   */
+  public validateStixObject<T extends StixObject>() {
+    // Return a function that takes an object and returns an Observable
+    return <P extends T>(object: P): Observable<any> => {
+      const url = `${this.apiUrl}/validate`;
+      const payload = {
+        type: object.type,
+        status: object.workflow ? object.workflow.state : 'work-in-progress',
+        stix: object.serialize().stix,
+      };
+      return this.http
+        .post(url, payload, {
+          headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
```

Comment:
Minor: you don't need to set the `'Content-Type': 'application/json'` header here - our [`AuthInterceptor`](https://github.com/center-for-threat-informed-defense/attack-workbench-frontend/blob/main/src/app/services/helpers/auth.interceptor.ts) adds it automatically to all requests